### PR TITLE
SD 1782: Provide a way to configure field name aliases for RSS ingest feeds.

### PR DESCRIPTION
**Pull request by plamut**
Provide a way to configure field name aliases for RSS ingest feeds.

### CODE REVIEW

File: server/superdesk/io/rss.py [hunk 1]
```diff
@@ -104,23 +127,32 @@ def _fetch_data(self, config, provider):
...
         :return: created content item
         :rtype: dict
         """
-        item = dict()
-        item['guid'] = item['uri'] = data.get('guid')
-        item['type'] = 'text'
-        if data.get('updated_parsed'):
-            item['versioncreated'] = utcfromtimestamp(timegm(data.get('updated_parsed')))
-        if data.get('published_parsed') or data.get('updated_parsed'):
-            item['firstcreated'] = utcfromtimestamp(timegm(data.get('published_parsed', data.get('updated_parsed'))))
```
Comment (plamut) [inline server/superdesk/io/rss.py]: @amagdas This behavior has slightly changed, the `firstcreated` field is now only generated from the `published_parsed` field in data, but not from `updated_parsed` - is that OK now that we can define field name aliases?

**Comment by plamut** [comment]
*Note to QA:* there is a bug with saving the provider configuration object to database, it is not possible to remove an already existing field alias even if correct data is sent in the PATCH request. Please keep this in mind when testing the feature.

(the bug itself has been reported in a separate issue)

**Comment by amagdas** [comment]
Unfortunately I have to :-1:  this PR, the downsides  of doing this are:
- much harder debugging for Ingest errors as the fields now are going to be provider dependent
- easy to introduce "Fat fingers" mistakes while configuring the ingest, after that we will have a bunch of data to fix
- the ingest channel configuration becomes tedious work as in order to get a working channel a bunch of work needs to be done

I would much rather provide some sane defaults and then allow for this aliasing if it's really needed.

**Comment by ioanpocol** [comment]
@amagdas "I would much rather provide some sane defaults and then allow for this aliasing if it's really needed. "
Yes, this was the idea and in this way it has been implemented.

**Comment by ioanpocol** [comment]
:+1:

**Comment by plamut** [comment]
@amagdas As for the complexitiy, it's true, aliases make debuggingmore difficult. OTOH is there any other sensible way to deal with _generic_ data sources? After some discussions we decided that aliases would be a good way to tackle this field naming problem.

As @ioanpocol has said, we do use default field names for data sources and aliases only need to be defined when a data source names a particular field differently (and certainly not for all the fields, that would indeed be a hassle).

**Comment by petrjasek** [comment]
:+1:

**Comment by amagdas** [comment]
Then please take care when merging this that all the current ingest sources work as expected on master and data instances. Probably we need to stop all RSS ingest feeds before merging, then do the needed configuration and then start the feed again.

**Comment by ioanpocol** [comment]
@amagdas Please confirm that your vote has been changed or not

**Comment by amagdas** [comment]
:+1: But please take care of properly handling the already configured rss providers on the running instances, if not celery will spit a lot of errors again.

**Comment by plamut** [comment]
Who has a good overveiw of all current RSS providers configured across all the instances we use?
(I don't, unfortunately)

**Comment by plamut** [comment]
Anyone? :)

**Comment by amagdas** [comment]
I will take care of merging this after I fix SD-1945

**Comment by plamut** [comment]
OK, sounds like a plan. Thanks!

**Comment by mugurrus** [comment]
:+1:

**Comment by amagdas** [comment]
Closing this, created #309 as a replacement.

**Pull request closed without merging**